static/js/src/base/navigation.js

```diff
@@ -87,4 +100,38 @@ document.addEventListener("click", function (event) {
   }
 });
 
+// Enable sticky navigation
+const enableStickyNav = () => {
+  document.addEventListener("DOMContentLoaded", function () {
+    var selector = "[data-js='sticky-nav-observer']";
+    // select the observer element
+    var observerEl = document.querySelector(selector);
+    // select the navigation element
+    var nav = document.getElementById("navigation");
+
+    if (observerEl && nav) {
+      // create a new observer
+      var observer = new IntersectionObserver(
+        function (entries) {
+          // add "sticky" class if the observerEl is not on the screen
+          if (entries[0].intersectionRatio === 0) {
+            nav.classList.add("is-sticky");
+            // remove "sticky" class if the observerEl is on the screen - i.e. you scrolled all the way to the top of the page
+          } else if (entries[0].intersectionRatio === 1)
+            nav.classList.remove("is-sticky");
+        },
+        { threshold: [0, 1] }
+      );
+      // ask the observer to observe the position of the observerEl - i.e. if it's on/off screen
+      observer.observe(observerEl);
+    } else {
+      throw new Error(
+        `${selector ? selector : "#navigation"} is not a valid element!`
+      );
+    }
+  });
+};
```

Comment:
I think this is being transpiled with babel - is that right? ES5 and ES6 syntax is being mixed in this one function: the function itself is a `const` and uses an arrow function expression, but the body of the function contains the use of `var` and traditional function expressions, along with ES6 style string interpolation.

If it's being transpiled, it doesn't matter whether it's ES5 or ES6, but I think the file should be consistent one way or the other.

Reply:
Yes, it is!you are right. this code is transpiled, but on juju.is is not. I'll have to amend the file there. I didn't see the elephant in the room 💯